gtsfm/multi_view_optimizer.py

```diff
@@ -28,118 +29,130 @@ def __init__(
         )
 
-        wti_graph, ta_metrics = self.trans_avg_module.create_computation_graph(
-            num_images,
-            pruned_i2Ui1_graph,
-            delayed_wRi,
-            absolute_pose_priors,
-            relative_pose_priors,
-            gt_wTi_list=gt_wTi_list,
-        )
-        init_cameras_graph = dask.delayed(init_cameras)(delayed_wRi, wti_graph, all_intrinsics)
+        if self._use_pose_slam_initialization:
+            delayed_poses, pose_slam_metrics = self.pose_slam_module.create_computation_graph(
+                num_images, relative_pose_priors=relative_pose_priors, gt_wTi_list=gt_wTi_list
+            )
+            metrics.append(pose_slam_metrics)
+        else:
+            delayed_wRi, rot_avg_metrics = self.rot_avg_module.create_computation_graph(
+                num_images, pruned_i2Ri1s, relative_pose_priors=relative_pose_priors, gt_wTi_list=gt_wTi_list
+            )
+
+            delayed_poses, ta_metrics = self.trans_avg_module.create_computation_graph(
+                num_images,
+                pruned_i2Ui1s,
+                delayed_wRi,
+                absolute_pose_priors,
+                relative_pose_priors,
+                gt_wTi_list=gt_wTi_list,
+            )
+            metrics += [rot_avg_metrics, ta_metrics]
 
-        ba_input_graph, data_assoc_metrics_graph = self.data_association_module.create_computation_graph(
+        initialized_cameras = dask.delayed(init_cameras)(delayed_poses, all_intrinsics)
+
+        ba_input, da_metrics = self.data_association_module.create_computation_graph(
             num_images,
-            init_cameras_graph,
-            viewgraph_v_corr_idxs_graph,
-            keypoints_graph,
+            initialized_cameras,
+            viewgraph_v_corr_idxs,
+            delayed_keypoints,
             cameras_gt,
             relative_pose_priors,
             images_graph,
         )
 
-        ba_result_graph, ba_metrics_graph = self.ba_optimizer.create_computation_graph(
-            ba_input_graph, absolute_pose_priors, relative_pose_priors, cameras_gt
+        ba_output, ba_metrics = self.ba_optimizer.create_computation_graph(
+            ba_input, absolute_pose_priors, relative_pose_priors, cameras_gt
         )
 
-        multiview_optimizer_metrics_graph = [
-            viewgraph_estimation_metrics,
-            rot_avg_metrics,
-            ta_metrics,
-            data_assoc_metrics_graph,
-            ba_metrics_graph,
+        metrics += [
+            da_metrics,
+            ba_metrics,
         ]
 
         # align the sparse multi-view estimate before BA to the ground truth pose graph.
-        ba_input_graph = dask.delayed(ba_input_graph.align_via_Sim3_to_poses)(gt_wTi_list)
+        # TODO(Frank): Why would we do this here? But maybe we should simply fix align_via_Sim3_to_poses
+        # ba_input_graph = dask.delayed(ba_input_graph.align_via_Sim3_to_poses)(gt_wTi_list)
```

Comment:
remove this todo?

Reply:
I think this is still left to act on.

Reply:
could you check who consumes ba_input? and whether they expect it to be aligned to GT? is the output aligned to GT?

Reply:
We are realigning it to GT in the visualization code. I think we need to fix whether the BA module returns an aligned result or not? BA has to align `GtsfmData` to evaluate metrics, but the run() function does not have the ground truth and does not perform the alignment. I think we can safely remove this commented out code now

Reply:
Removed the commented out code for now as we are doing alignment after this function too.